src/NuGet.Core/NuGet.PackageManagement/Audit/AuditCheckResult.cs

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using NuGet.Common;
+using NuGet.Packaging.Core;
+
+namespace NuGet.PackageManagement
+{
+    public record AuditCheckResult
+    {
+        public IReadOnlyList<ILogMessage> Warnings { get; }
+        internal bool IsAuditEnabled { get; set; } = true;
```

Comment:
> From a public API design point of view, personally I don't like that this is a public class, and therefore part of the public API, but with properties that other assemblies can't set. It makes it impossible to use in unit tests for a start, except for assemblies that this assembly lists as InternalsVisibleTo.
If the intention is that this type should only be created from within the assembly, never by others (even for unit tests), then I suggest adding an internal constructor, so it doesn't have a public constructor.

I'll change it to an internal constructor.

> But if this type needs to be public to be able to get the warnings and set the telemetry, then I think the properties should be public too. As I mentioned in another comment, the properties can change set to init, to prevent tampering, making it a "read only" (immutable, depending on your definition of immutable) type.

I feel this overexposes internals that are rather flexible. Metrics we'd want to collect would change but they don't fundamentally change the result of the vulnerability check.